config-scalecube-configuration-service/src/main/java/io/scalecube/config/service/ScalecubeConfigurationServiceConfigSource.java

```diff
@@ -0,0 +1,133 @@
+import com.fasterxml.jackson.databind.ObjectWriter;
+import io.scalecube.config.ConfigProperty;
+import io.scalecube.config.ConfigSourceNotAvailableException;
+import io.scalecube.config.source.ConfigSource;
+import io.scalecube.config.source.LoadedConfigProperty;
+import io.scalecube.config.utils.ThrowableUtil;
+import io.scalecube.configuration.api.ConfigurationService;
+import io.scalecube.configuration.api.EntriesRequest;
+import io.scalecube.configuration.api.FetchResponse;
+import io.scalecube.services.gateway.clientsdk.ClientSettings;
+import io.scalecube.services.gateway.clientsdk.ClientSettings.Builder;
+import io.scalecube.services.transport.jackson.JacksonCodec;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.netty.http.HttpResources;
+
+public class ScalecubeConfigurationServiceConfigSource implements ConfigSource {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(ScalecubeConfigurationServiceConfigSource.class);
+
+  private final ConfigurationService service;
+
+  private final EntriesRequest requestEntries;
+
+  private final Parsing parsing = new Parsing();
+
+  public static class Builder {
+    private String token;
+    private String repository;
+    private URL url = ScalecubeConfigurationServiceConfigSource.defaultUrl();
+    private ConfigurationService service;
+
+    public Builder() {}
+
+    public ScalecubeConfigurationServiceConfigSource build() {
+      if (Objects.requireNonNull(this.token, "Missing token").isEmpty()) {
+        throw new IllegalArgumentException("Missing token");
+      }
+      if (Objects.requireNonNull(this.repository, "Missing repository").isEmpty()) {
+        throw new IllegalArgumentException("Missing repository");
+      }
+      if (this.service == null) {
+        ClientSettings.Builder builder =
+            ClientSettings.builder()
+                .host(url.getHost())
+                .port(url.getPort())
+                .contentType(JacksonCodec.CONTENT_TYPE)
+                .loopResources(HttpResources.get());
+        if ("https".equals(url.getProtocol()) || "wss".equals(url.getProtocol())) {
+          builder = builder.secure();
+        }
+        this.service = http(builder.build()).forService(ConfigurationService.class);
+      }
+      return new ScalecubeConfigurationServiceConfigSource(this);
+    }
+
+    public Builder token(String token) {
+      this.token = token;
+      return this;
+    }
+
+    public Builder repository(String repository) {
+      this.repository = repository;
+      return this;
+    }
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  private static URL defaultUrl() {
+    try {
+      return new URL("https", "configuration-service-http.genesis.om2.com", 443, "/");
```

Comment:
Let's do not hardcode anything that related to specific (internal) deployment infrastructure into an open source project.